### PR TITLE
wire printflags through create commands

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

~~Depends on https://github.com/kubernetes/kubernetes/pull/62139~~
~~Depends on https://github.com/kubernetes/kubernetes/pull/62341~~

Begins wiring PrintFlags through create subcommands.
Opening now to begin gathering feedback.

cc @soltysh @deads2k

### CODE REVIEW

File: pkg/kubectl/cmd/create.go [hunk 1]
```diff
@@ -295,17 +315,51 @@ func NameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
...
 	// Name of resource being created
 	Name string
 	// StructuredGenerator is the resource generator for the object being created
 	StructuredGenerator kubectl.StructuredGenerator
 	// DryRun is true if the command should be simulated but not run against the server
-	DryRun       bool
-	OutputFormat string
+	DryRun           bool
+	CreateAnnotation bool
+	OutputFormat     string
```
Comment (soltysh) [inline pkg/kubectl/cmd/create.go]: `OutputFormat` is not needed anymore.

File: pkg/kubectl/cmd/create_clusterrole.go [hunk 2]
```diff
@@ -172,10 +176,5 @@ func (c *CreateClusterRoleOptions) RunCreateRole() error {
 		}
 	}
 
-	if useShortOutput := c.OutputFormat == "name"; useShortOutput || len(c.OutputFormat) == 0 {
-		cmdutil.PrintSuccess(useShortOutput, c.Out, clusterRole, c.DryRun, "created")
-		return nil
-	}
-
 	return c.PrintObject(clusterRole)
```
Comment (soltysh) [inline pkg/kubectl/cmd/create_clusterrole.go]: Use `PrintObj` for consistency with all the changes.

File: pkg/kubectl/cmd/create_deployment.go [hunk 3]
```diff
@@ -36,10 +37,25 @@ var (
...
 // This command is better known to users as `kubectl create deployment`.
 // Note that this command overlaps significantly with the `kubectl run` command.
 func NewCmdCreateDeployment(f cmdutil.Factory, cmdOut, cmdErr io.Writer) *cobra.Command {
+	options := &DeploymentOpts{
+		CreateSubcommandOptions: &CreateSubcommandOptions{
+			PrintFlags: printers.NewPrintFlags("created"),
+			CmdOut:     cmdOut,
+		},
+
+		CmdErr: cmdErr,
```
Comment (soltysh) [inline pkg/kubectl/cmd/create_deployment.go]: Put `CmdErr` into the common struct `CreateSubcommandOptions` even if not all use it, it still does make sense to be just there.

**Review by soltysh — CHANGES_REQUESTED** [review]
A few nits here and there, but mostly looks good.

**Comment by juanvallejo** [comment]
@soltysh thanks for the review, comments addressed.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve no-issue

**Pull request merged**